geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala

```diff
@@ -85,7 +86,12 @@ abstract class GeoMesaDataStore[DS <: GeoMesaDataStore[DS]](val config: GeoMesaD
     * (index tables and catalog table)
     * NB: We are *not* currently deleting the query table and/or query information.
     */
-  def delete(): Unit = adapter.deleteTables(getTypeNames.flatMap(getAllTableNames).distinct)
+  def delete(): Unit = {
+    val types = getTypeNames
+    val tables = if (types.length == 0) Array(config.catalog) else types
+    adapter.deleteTables(tables.flatMap(getAllTableNames).distinct)
+    metadata.asInstanceOf[TableBasedMetadata[String]].resetCache()
```

Comment:
this cast is unnecessary now, right?
```suggestion
    metadata.resetCache()
```